### PR TITLE
os/bluestore: Use Google Test fixtures to avoid duplicate code

### PR DESCRIPTION
**Pull request by agayev**
os/bluestore: Use Google Test fixtures to avoid duplicate code

Signed-off-by: Abutalib Aghayev <agayev@cs.cmu.edu>

### CODE REVIEW

**Comment by liewegas** [comment]
Can you put the signed-off-by line in the git commit message?

**Comment by agayev** [comment]
> Can you put the signed-off-by line in the git commit message?

Done.

File: src/test/objectstore/test_bluefs.cc [hunk 1]
```diff
@@ -247,135 +246,81 @@ void join_all(std::vector<std::thread>& v)
 #define NUM_SINGLE_FILE_WRITERS 1
 #define NUM_MULTIPLE_FILE_WRITERS 2
 
-TEST(BlueFS, test_flush_1) {
-  uint64_t size = 1048576 * 128;
-  string fn = get_temp_bdev(size);
-  g_ceph_context->_conf.set_val(
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: Looks like this is lost

File: src/test/objectstore/test_bluefs.cc [hunk 2]
```diff
@@ -247,135 +246,81 @@ void join_all(std::vector<std::thread>& v)
...
     join_all(write_thread_multiple);
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_flush_2) {
-  uint64_t size = 1048576 * 256;
-  string fn = get_temp_bdev(size);
-  g_ceph_context->_conf.set_val(
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: the same here

File: src/test/objectstore/test_bluefs.cc [hunk 3]
```diff
@@ -247,135 +246,81 @@ void join_all(std::vector<std::thread>& v)
...
 
     join_all(write_thread_single);
     join_all(write_thread_multiple);
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_flush_2) {
-  uint64_t size = 1048576 * 256;
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: you're changing the size with your patch. I'm not sure whether it's critical but your new infrastructure provides no means to adjust such stuff on-demand. This is definitely some flexibility loss and a bit of cheating.

File: src/test/objectstore/test_bluefs.cc [hunk 4]
```diff
@@ -247,135 +246,81 @@ void join_all(std::vector<std::thread>& v)
...
     join_all(write_thread_multiple);
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_flush_3) {
-  uint64_t size = 1048576 * 256;
-  string fn = get_temp_bdev(size);
-  g_ceph_context->_conf.set_val(
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: same here

File: src/test/objectstore/test_bluefs.cc [hunk 5]
```diff
@@ -247,135 +246,81 @@ void join_all(std::vector<std::thread>& v)
...
     }
 
     join_all(write_thread_multiple);
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_flush_3) {
-  uint64_t size = 1048576 * 256;
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: size change again

File: src/test/objectstore/test_bluefs.cc [hunk 6]
```diff
@@ -439,133 +373,89 @@ TEST(BlueFS, test_simple_compaction_async) {
...
+    fs->compact_log();
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_compaction_async) {
-  uint64_t size = 1048576 * 128;
-  string fn = get_temp_bdev(size);
-  g_ceph_context->_conf.set_val(
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: same here

File: src/test/objectstore/test_bluefs.cc [hunk 7]
```diff
@@ -439,133 +373,89 @@ TEST(BlueFS, test_simple_compaction_async) {
...
+    fs->compact_log();
   }
-  fs.umount();
-  rm_temp_bdev(fn);
 }
 
-TEST(BlueFS, test_replay) {
-  uint64_t size = 1048576 * 128;
-  string fn = get_temp_bdev(size);
-  g_ceph_context->_conf.set_val(
```
Comment (ifed01) [inline src/test/objectstore/test_bluefs.cc]: lost

**Pull request closed without merging**